PollBuddy-Server/frontend/src/pages/PollViewer/PollViewer.js

```diff
@@ -29,18 +29,14 @@ export default class PollViewer extends Component {
         } else {
           throw new Error("Something went wrong");
         }
-
       })
       .then(response => {
-        if (response === {}) {
-          console.log("Error fetching data");
-        } else {
-          console.log("Fetching data succeeded");
-          console.log(response);
+        if (response.result === "success") {
           this.setState({"questionData": response.data, "doneLoading": true});
+        } else {
+          window.location.href = "/login";
         }
-      })
-      .catch(error => this.setState({"error": error}));
```

Comment:
I think we'll need the catch too if we're throwing the error above. Also, you can keep the `if (response.result === "success") {`, that was good. I think we also talked about handling the actual error messages if `result == failure` too. You should be able to find those in the backend documentation or wiki pages.